ecosystem/node-checker/src/evaluators/direct/api/transaction_availability.rs

```diff
@@ -104,88 +75,125 @@ impl Evaluator for TransactionAvailabilityEvaluator {
     /// baseline produced after a delay. We confirm that the transactions are
     /// same by looking at the version.
     async fn evaluate(&self, input: &Self::Input) -> Result<Vec<EvaluationResult>, Self::Error> {
+        let oldest_baseline_version = input
+            .baseline_index_response
+            .ledger_info
+            .oldest_ledger_version
+            .0;
+        let oldest_target_version = input
+            .target_index_response
+            .ledger_info
+            .oldest_ledger_version
+            .0;
+        let latest_baseline_version = input.baseline_index_response.ledger_info.ledger_version.0;
+        let latest_target_version = input.target_index_response.ledger_info.ledger_version.0;
+
+        // Get the oldest ledger version between the two nodes.
+        let oldest_shared_version = min(oldest_baseline_version, oldest_target_version);
+
+        // Get the least up to date latest ledger version between the two nodes.
+        let latest_shared_version = min(latest_baseline_version, latest_target_version);
+
+        // Ensure that there is a window between the oldest shared version and
+        // latest shared version. If there is not, it will not be possible to
+        // pull a transaction that both nodes have.
+        if oldest_shared_version > latest_shared_version {
+            return Ok(vec![self.build_evaluation_result(
+                "Unable to pull transaction from both nodes".to_string(),
+                50,
```

Comment:
nit: I'd probably give 0 (or 25) here given that we have such large values for pruning windows. If someone hits this case, I think it's really odd and likely signifies they're out of data (or the baseline is broken).